collect_repos.py

```diff
@@ -0,0 +1,75 @@
+'''
+This script collects the current state of all student repositories for a given assignment in a given organization 
+Run this script from inside a directory that contains/will contain all the repos for this assignment
+'''
+
+from github3 import login
+import getpass
+import subprocess
+import csv
+import configparser
+import os
+
+def run(username, token, organization, prefix, list, deadline):
+	with open(token, 'r') as f:
+		token = f.readline().strip()
+
+	github = login(username,token=token)
+	memberships = github.organization_memberships()
+	course = None
+
+	for membership in memberships:
+		if membership.organization.login.lower() == organization.lower():
+			course = membership.organization
+			break
+
+	newDir = os.getcwd() + "/{}".format(prefix) #pull all repos to this folder
+	if not os.path.exists(newDir):
+		os.mkdir(newDir)
+
+	with open('student_list.csv', mode='r') as f:
+		reader = csv.reader(f)
+		mapping = {rows[1]:rows[2] for rows in reader}
+
+	if list:
+		with open(list, 'r') as f:
+			ccids = [ccid.strip() for ccid in f.readlines()]
+			usernames = [mapping[ccid] for ccid in ccids]  #convert ccid to github username
+	else:
+		usernames = []
+
+	repos = []
+	for repo in course.repositories():
+		if len(usernames):
+			if repo.name.replace('{}-'.format(prefix), '') in usernames:
```

Comment:
Will this match only the first occurrence of the -?

Reply:
Even if a student has a Github username "labs-abc", his repo name is "labs-labs-abc".
After the replace method, the repo name becomes "abc", but "abc" doesn't match his username "labs-abc".

Reply:
ahh nvm, you are looking for "labs-", i was thinking of only - and a lot of usernames have - in them. Should be ok then.
In your example, I think the repo name should become "labs-abc" to reflect the right username but I think that's a bit of an extreme case we can ignore for now